File: cs/Iris/ScriptElements/NpcStructure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Iris.ExpressionElement;

namespace Iris.ScriptElements
{
    public class NpcStructure
    {
        public string name;

        //public List<LabelExpression> labels;
        public Dictionary<string, Expression> labels = new Dictionary<string, Expression>();

        public NpcStructure()
        {

        }

        public NpcStructure(string npcName)
        {
            name = npcName;
        }

        public void addLabel(LabelExpression exp)
        {
            addLabel(exp.labelName, exp.body);
        }

        public void addLabel(string labelName, Expression exp)
        {
            if (labels.ContainsKey(labelName) || exp == null)
            {
                return;
            }

            labels.Add(labelName, exp);
        }

        public bool validNpc()
        {
            return labels.ContainsKey("main");
        }
    }
}

File: java/Iris/src/com/rokuan/iris/script/NpcStructure.java
package com.rokuan.iris.script;

import java.util.ArrayList;
import java.util.HashMap;

import com.rokuan.iris.expression.Expression;
import com.rokuan.iris.expression.LabelExpression;

public class NpcStructure
{
	public String name;
	public HashMap<String, Expression> labels = new HashMap<String, Expression>();
	public String location;
	public ArrayList<Integer> coordinates = new ArrayList<Integer>();

	public NpcStructure()
	{

	}

	public NpcStructure(String npcName)
	{
		name = npcName;
	}

	public void addLabel(LabelExpression exp)
	{
		addLabel(exp.labelName, exp.body);
	}

	public void addLabel(String labelName, Expression exp)
	{
		if (labels.containsKey(labelName) || exp == null)
		{
			return;
		}

		labels.put(labelName, exp);
	}
	
	public void addCordinate(int coord){
		coordinates.add(coord);
	}

	public boolean validNpc()
	{
		return labels.containsKey("main");
	}
}